nodejs/awsx/apigateway/README.md

```diff
@@ -202,6 +202,242 @@ let endpoint = new awsx.apigateway.API("example", {
+            };
+        },
+        authorizers: [{
+            parameterName: "Authorization",
+            parameterLocation: "header",
+            authType: "oauth2",
+            type: "request",
+            handler: async (event: awsx.apigateway.AuthorizerEvent) => {
+                // Add your own custom authorization logic here.
+                 return awsx.apigateway.AuthorizerResponse(
+                    "user",
+                    "Allow",
+                    event.methodArn);
+            },
+        }],
+    }],
+});
+```
+
+#### Specifying the Role
+
+If your Authorizer requires access to other AWS resources, you will need to provision the appropriate role. You can do so by using `new aws.lambda.CallbackFunction`.
+
+```ts
+import * as aws from "@pulumi/aws";
+import * as awsx from "@pulumi/awsx";
+
+const callbackFxn = new aws.lambda.CallbackFunction("callbackFxn", {
+    callback: async (event: awsx.apigateway.AuthorizerEvent) => {
+        // Add custom authorization logic here
+        return awsx.apigateway.AuthorizerResponse(
+            "user",
+            "Allow",
+            event.methodArn);
+    },
+    role: role, // Specify role with appropriate AWS permissions.
+});
+
+const api = new awsx.apigateway.API("myapi", {
+    routes: [{
+        path: "/b",
+        method: "GET",
+        eventHandler: async () => {
+            return {
+                statusCode: 200,
+                body: "<h1>Hello world!</h1>",
+            };
+        },
+        authorizers: [{
+            parameterName: "Authorization",
+            parameterLocation: "header",
+            authType: "oauth2",
+            type: "request",
+            handler: callbackFxn,
+        }],
+    }],
+});
+```
+
+#### Using a Lambda Predefined Lambda
+
+You can also define the Lambda Authorizer elsewhere and then reference the required values.
+
+```ts
+import * as awsx from "@pulumi/awsx";
+
+const apiWithAuthorizer = new awsx.apigateway.API("authorizer-api", {
+    routes: [{
+        ...
+        authorizers: [{
+            authorizerName: "testing",
+            parameterName: "auth",
+            parameterLocation: "query",
+            authType: "custom",
+            type: "request",
+            handler: {
+                // Either specify the aws.lambda.Function or provide the invoke URI
+                uri: authorizerLambda,
+                credentials: gatewayRole.arn,
+            },
+            identitySource: ["method.request.querystring.auth"],
+        }],
+    }],
+});
+```
+
+A complete example of defining the Lambda Authorizer elsewhere can be found [here](https://github.com/pulumi/pulumi-awsx/blob/61d2996b8bdb20ea625e66e17ebbaa7b62f9c163/nodejs/awsx/examples/api/index.ts#L94-L152).
+
+#### Authorizer Response
+
+A helper function `awsx.apigateway.AuthorizerResponse` has been created to simplify generating the authorizer response. This can be used when defining the authorizer handler as follows:
```

Comment:
if it's a function, we should camelCase it.

Reply:
also, i'm not sure we need tihs.  you already use this helper above, so the readme naturally points people toward using this helper.